src/ring_theory/adjoin_root.lean

```diff
@@ -416,4 +416,86 @@ end field
+  for `α` a root of `f : polynomial R` and `I : ideal R`-/
+def quot_map_C_map_span_mk_equiv_quot_map_C_quot_map_span_mk : (adjoin_root f) ⧸
+  (((I.map (C : R →+* polynomial R))).map (span ({f} : set (polynomial R)))^.quotient.mk) ≃+*
+  (polynomial R ⧸ map C I) ⧸ ((span ({f} : set (polynomial R))).map
+  (I.map (C : R →+* polynomial R))^.quotient.mk) :=
+quot_quot_equiv_comm (span ({f} : set (polynomial R))) (I.map (C : R →+* polynomial R))
+
+lemma quot_map_C_map_span_mk_equiv_quot_map_C_quot_map_span_mk_mk (p : polynomial R) :
+  quot_map_C_map_span_mk_equiv_quot_map_C_quot_map_span_mk I f (ideal.quotient.mk _ (mk f p)) =
+  quot_quot_mk (I.map (C : R →+* polynomial R)) (span ({f} : set (polynomial R))) p := rfl
+
+/-- The natural isomorphism `(R[x]/I[x])/((f) ⊔ I[x] / I[x]) ≅ (R[x]/I*R[x])/(f mod I[x])` where
+  `f : polynomial R` and `I : ideal R`-/
+def quot_map_C_quot_map_span_mk_equiv_quot_map_C_quot_mk_span : (polynomial R ⧸ map C I) ⧸
+  ((span ({f} : set (polynomial R))).map (I.map (C : R →+* polynomial R))^.quotient.mk) ≃+*
+  (polynomial R ⧸ map C I) ⧸
+  (span ({(ideal.quotient.mk (I.map polynomial.C)) f} : set (polynomial R ⧸ map C I))) :=
+ideal.quot_equiv_of_eq (by rw [map_span, set.image_singleton])
+
+lemma quot_map_C_quot_map_span_mk_equiv_quot_map_C_quot_mk_span_quot_quot_mk (p : polynomial R) :
+  quot_map_C_quot_map_span_mk_equiv_quot_map_C_quot_mk_span I f (quot_quot_mk _ _ p) =
+  ideal.quotient.mk _ (ideal.quotient.mk _ p) := rfl
+
+/-- the natural isomorphism `(R/I)[x]/(f mod I) ≅ (R[x]/I*R[x])/(f mod I[x])` where
+  `f : polynomial R` and `I : ideal R`-/
+def polynomial_over_quot_quot_mk_span_equiv_quot_map_C_quot_mk_span : polynomial (R ⧸ I) ⧸
+  (span ({f.map (I^.quotient.mk)} : set (polynomial (R ⧸ I)))) ≃+*
+  (polynomial R ⧸ map C I) ⧸
+  (span ({(ideal.quotient.mk (I.map polynomial.C)) f} : set (polynomial R ⧸ map C I))) :=
+quotient_equiv (span ({f.map (I^.quotient.mk)} : set (polynomial (R ⧸ I))))
+  (span {ideal.quotient.mk (I.map polynomial.C) f})
+  (polynomial_quotient_equiv_quotient_polynomial I)
+  (by rw [map_span, set.image_singleton, ring_equiv.coe_to_ring_hom,
+    polynomial_quotient_equiv_quotient_polynomial_map_mk I f])
+
+lemma polynomial_over_quot_quot_mk_span_equiv_quot_map_C_quot_mk_span_symm_mk_mk (p : polynomial R):
+  (polynomial_over_quot_quot_mk_span_equiv_quot_map_C_quot_mk_span I f).symm
+  (ideal.quotient.mk _ (ideal.quotient.mk _ p)) =
+  (ideal.quotient.mk  _ (p.map I^.quotient.mk)) :=
+by simp only [polynomial_over_quot_quot_mk_span_equiv_quot_map_C_quot_mk_span,
+  quotient_equiv_symm_mk, polynomial_quotient_equiv_quotient_polynomial_symm_mk]
+
+/-- The natural isomorphism `R[α]/I[α] ≅ (R/I)[X]/(f mod I)` for `α` a root of `f : polynomial R`
+  and `I : ideal R`-/
+def quot_adjoin_root_equiv_quot_polynomial_quot : (adjoin_root f) ⧸ (I.map (of f)) ≃+*
+polynomial (R ⧸ I) ⧸ (span ({f.map (I^.quotient.mk)} : set (polynomial (R ⧸ I)))) :=
```

Comment:
Do you agree that this is the "real" equivalence we're interested in, and the others are helpers for constructing it? Then this deserves a short name, and perhaps some docstrings pointing from the others to this one.
```suggestion
def quot_map_of_equiv :
  adjoin_root f ⧸ I.map (of f) ≃+* (R ⧸ I)[X] ⧸ span {f.map (I^.quotient.mk)} :=
```